vwifi.c

```diff
@@ -846,17 +846,102 @@ static int owl_delete_interface(struct owl_vif *vif)
     return 0;
 }
 
+/* Called by kernel when user decided to change the interface type. */
+static int owl_change_iface(struct wiphy *wiphy,
+                            struct net_device *ndev,
+                            enum nl80211_iftype type,
+                            struct vif_params *params)
+{
+    switch (type) {
+    case NL80211_IFTYPE_STATION:
+    case NL80211_IFTYPE_AP:
+        ndev->ieee80211_ptr->iftype = type;
+        break;
+    default:
+        pr_info("owl: invalid interface type %u\n", type);
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+/* Called by the kernel when user want to create an Access Point. Now
+ * it just add a ssid to the ssid_table to emulate the AP signal. And
+ * record the ssid to the owl_context.
+ */
+static int owl_start_ap(struct wiphy *wiphy,
+                        struct net_device *ndev,
+                        struct cfg80211_ap_settings *settings)
+{
+    struct owl_vif *owl = ndev_get_owl_vif(ndev);
```

Comment:
The variable name should `vif` rather than `owl`, since `owl` represent `struct owl_conext`.

Reply:
ok